### DISCUSSION TITLE
findAllListingsActive endpoint has offset limit?

### DISCUSSION THREAD

**Discussion by myth-ila** [etsy/open-api#952] [Q&A] [answered]
Hi all,
I am trying to extract all active listings under a particular taxonomy category .
While I am able to extract listings using the api, I repeatedly get a HTTP 400 error after the offset parameter reaches 12100.
Does this endpoint have an offset limit? how do I go around this?

Thanks in advance!

**Comment by Gareth064** [comment]
I have seen a couple of different issues mentioned by other users over the past couple of years.

One where the filtering on that endpoint wasn't applying, not sure if it was taxonomy related exactly

Then another where the pagination worked for the first offset or so, but then it started to return unrelated listings to what the filter was asking for.

Not sure if these were ever fixed or not.

When you get a response from a request, the headers include an Etsy Request ID. If you grab that from a request which doesn't return what you expected then you should email the Etsy Devs, explain the problem and provide the Etsy Request IDs from your requests and they should be able to trace what happens at their end.

**Reply by myth-ila** [reply]
Sure! thanks for the suggestion! I will reach out to Etsy Devs. Any idea on the turnaround time for the response? Just want to get a sense of that since its my first time doing that.

**Reply by Gareth064** [reply]
Sorry, no idea how long they might take to respond. It's sporadic to say the least.

**Comment by ablagoev** [comment, marked as answer]
Hey, additionally there is a hard limit for the offset parameter to 12000. You can find this in the docs here:

https://developers.etsy.com/documentation/essentials/urlsyntax#pagination

```
The minimum and default records returned per call is 25, and the maximum number that can be returned is 100. We provide limit and offset parameters to allow navigation through larger data sets. Responses include a count field, which specifies the total number of records available via pagination. For performance reasons, the offset parameter is limited to a maximum value of 12000.
```

Depending on your use case you may need to narrow down your filters. I.e. fetching listings shop by shop, etc. so that you do not hit the limit.

Hope this helps.

Cheers,
Alex

**Reply by myth-ila** [reply]
Hi Alex,
Thanks for this. I seem to have missed this point when I looked at the documentation. I actually need to get a list of all shops but unfortunately I couldn't find a getallshops kind of endpoint. So, I was using this round about way of getting all listings and then getting to a list of shops . Do you know of any way to get an exhaustive list of shops?
Best,
Mithila

**Reply by Gareth064** [reply]
All there is is the findShops endpoint but you need to provide a search term which searches against the name of the shops.
